### PR TITLE
[statsd-emitter] Fix service tag by avoiding using a reserved word

### PR DESCRIPTION
**Pull request by AlexandreYang**
### Description

The `service` tag introduced in this PR https://github.com/apache/incubator-druid/pull/8238 is a reserved word: https://docs.datadoghq.com/tagging/

Hence it can't be used by `statsd-emitter` extension. This PR fix the issue by renaming the tag to `druid_service` (underscore is more suitable than camelCase since since Datadog lowercase the tag names).

#### Fixed the bug ...
Fix a bug introduced in https://github.com/apache/incubator-druid/pull/8238

#### Added a forbidden-apis entry ...

<hr>

This PR has:
- [x] been self-reviewed.
- [X] added unit tests or modified existing tests to cover new code paths.

<hr>

### CODE REVIEW

File: extensions-contrib/statsd-emitter/src/test/java/org/apache/druid/emitter/statsd/StatsDEmitterTest.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ public void testServiceAsTagOption()
             client
     );
     client.time("druid.query.time", 10,
-            "service:druid/broker", "dataSource:data-source", "type:groupBy", "hostname:brokerHost1"
+            "druidService:druid/broker", "dataSource:data-source", "type:groupBy", "hostname:brokerHost1"
```
Comment (ccaominh) [inline extensions-contrib/statsd-emitter/src/test/java/org/apache/druid/emitter/statsd/StatsDEmitterTest.java]: Is there any way to test that reserved tag keys are not used?

Comment (AlexandreYang) [reply extensions-contrib/statsd-emitter/src/test/java/org/apache/druid/emitter/statsd/StatsDEmitterTest.java]: The list of reserved is on this documentation: https://docs.datadoghq.com/tagging/

Testing that would mean adding the list of reserve words in this project, we will need to maintain that list to match the "source of truth" that is elsewhere (the doc above). So I'm not sure it worth it. What do you think ?

Comment (ccaominh) [reply extensions-contrib/statsd-emitter/src/test/java/org/apache/druid/emitter/statsd/StatsDEmitterTest.java]: I was wondering if there was an easy way to automatically get the list of reserved tag keys, but if not, then I don't think it's worth manually syncing something to the source of truth.

Comment (AlexandreYang) [reply extensions-contrib/statsd-emitter/src/test/java/org/apache/druid/emitter/statsd/StatsDEmitterTest.java]: AFAIK, there is no such list of tags we can import easily in this project :(

**Pull request merged**